File: src/longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        lcp = ''

        while True:
            for i in range(len(strs)):
                if len(strs[i]) <= len(lcp) or strs[i][len(lcp)] != strs[0][len(lcp)]:
                    return lcp
            lcp+=strs[0][len(lcp)]

        return lcp

File: src/longest-common-prefix/longest-common-prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  lcp = ""

  loop do
    ch = strs[0][lcp.length]

    strs.each do |str|
      return lcp if str.length <= lcp.length || str[lcp.length] != ch
    end

    lcp += ch
  end

  lcp
end
